static/app/components/notificationActions/notificationActionManager.tsx

```diff
@@ -0,0 +1,203 @@
+import {Fragment, useState} from 'react';
+import capitalize from 'lodash/capitalize';
+
+import DropdownButton from 'sentry/components/dropdownButton';
+import {DropdownMenu, MenuItemProps} from 'sentry/components/dropdownMenu';
+import NotificationActionItem from 'sentry/components/notificationActions/notificationActionItem';
+import {IconAdd} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import {Project} from 'sentry/types';
+import {
+  AvailableNotificationAction,
+  NotificationAction,
+  NotificationActionService,
+} from 'sentry/types/notificationActions';
+
+type NotificationActionManagerProps = {
+  /**
+   * The list of existing notification actions
+   */
+  actions: NotificationAction[];
+  /**
+   * The list of available notification actions
+   */
+  availableActions: AvailableNotificationAction[];
+  /**
+   * The project associated with the notification actions
+   * TODO(enterprise): refactor to account for multiple projects
+   */
+  project: Project;
+  /**
+   * Updates the notification alert count for this project
+   */
+  updateAlertCount: (projectId: number, alertCount: number) => void;
+  /**
+   * Optional list of roles to display as recipients of Sentry notifications
+   */
+  recipientRoles?: string[];
+};
+
+const NotificationActionManager = ({
+  actions,
+  availableActions,
+  recipientRoles,
+  project,
+  updateAlertCount,
+}: NotificationActionManagerProps) => {
+  const [notificationActions, setNotificationActions] =
+    useState<Partial<NotificationAction>[]>(actions);
+
+  const addNotificationAction = (
+    actionInstance: AvailableNotificationAction['action']
+  ) => {
+    const updatedActions = [...notificationActions, actionInstance];
+    setNotificationActions(updatedActions);
+    updateAlertCount(parseInt(project.id, 10), updatedActions.length);
+  };
+
+  const removeNotificationAction = (index: number) => {
+    // Removes notif action from state using the index
+    const updatedActions = [...notificationActions];
+    updatedActions.splice(index, 1);
+    setNotificationActions(updatedActions);
+    updateAlertCount(parseInt(project.id, 10), updatedActions.length);
+  };
+
+  const updateNotificationAction = (index: number, updatedAction: NotificationAction) => {
+    // Updates notif action from state using the index
+    const updatedActions = [...notificationActions];
+    updatedActions.splice(index, 1, updatedAction);
+    setNotificationActions(updatedActions);
+  };
+
+  // Lists the available actions for each service
+  const availableServices: Record<
+    NotificationActionService,
+    AvailableNotificationAction[]
+  > = {
+    [NotificationActionService.SENTRY_NOTIFICATION]: [],
+    [NotificationActionService.EMAIL]: [],
+    [NotificationActionService.SLACK]: [],
+    [NotificationActionService.PAGERDUTY]: [],
+    [NotificationActionService.MSTEAMS]: [],
+    [NotificationActionService.SENTRY_APP]: [],
+  };
+  availableActions.forEach(a => {
```

Comment:
🤔 

useEffect here?
Or memoize?

Reply:
this doesn't get updated unless somebody goes to a different page and adds integrations